queries/objdump/highlights.scm

```diff
@@ -0,0 +1,20 @@
+(address) @number
+(byte) @constant
+(hexadecimal) @number
+(integer) @number
+(section_address) @number
+
+(disassembly_section_label) @text.title
+[(file_offset) (discriminator)] @field
+(file_path) @string
+(identifier) @namespace
+(instruction) @text
```

Comment:
```suggestion
(instruction) @function
```

Reply:
Sometimes the `(instruction)` has garbage contents inside of it when objdump cannot read the instruction. For example it'll literally write `(bad)` instead of a real instruction like `pop rbx`. Do you still want `@function` in those cases or would you like me to add those exceptional cases as separate highlights to `[..."File Offset"...] @text`?

Reply:
`pop` should be `@function`
`rbx` should be `@variable.builtin` (maybe)
`(bad)` should be `@text.warning`

Reply:
At the moment the objdump parser doesn't attempt to parse inner assembly instructions like common x86 opcodes and registers like `pop` and `rbx`. We could have common instruction highlights even without explicitly adding parse rules but I was kind of hoping injections could solve that problem for me, since https://github.com/rush-rs/tree-sitter-asm and https://github.com/bearcove/tree-sitter-x86asm already exist.

Admittedly I haven't looked into whether injections are up to that task just yet. If it's important for a v1 to sort this now, I can start on that, otherwise I'd like to defer that decision to a v2.

If it's alright, for now though I can stick with `(instruction)` == `@function` and `(bad)` == `@text.warning`. If there's other cases I'll try to collect them and mark them separately too.

Reply:
`(instruction)` from `@text` -> `@function` is done

Reply:
I've added parse + highlight support for (bad). Please let me know if we're good to resolve this or if I need to start looking at assembly parsing (via injections or otherwise)

Reply:
Then maybe go with `@text.literal` for instructions?

Reply:
> Then maybe go with @text.literal for instructions?

Done!

Regarding assembly highlights - It sounds like, according to https://github.com/nvim-treesitter/nvim-treesitter/pull/5548#issuecomment-1773707396, that we should be able to rely on injections after all. So I'm going to resolve this conversation but please unresolve if you think there's more things left to do here